### ISSUE TITLE
Filter functionality on admin pages don't work unless actions = None on ModelAdmins

### ISSUE THREAD

**Issue by karansag** [sehmaschine/django-grappelli#214]
Hi all,

Unless actions = None in my ModelAdmin class definitions, grappelli raises and javascript exception and the filter functionality (with fields defined in list_filter) breaks. The dropdown button shows up but nothing drops down when you click on it.

The error is "Uncaught TypeError: Object [object Object] has no method 'actions' " on the line $("tr input.action-select").actions(); (line 66 in the grappelli template).

The relevant code snippet from the grappelli code

<script type="text/javascript" charset="utf-8">
            (function($) {
                $(document).ready(function() {
                    $("tr input.action-select").actions();

                });
            })(grp.jQuery);
        </script>

Also, I think this might be similar to Issue #206. 

I'm using Django 1.4.1 and Grappelli 2.4.2

**Comment by sdillinger** [comment]
Same problem that's been reported for 2+ years.   Tried all reasonable solutions.  No luck.

**Comment by sehmaschine** [comment]
@sdillinger well, haven't been able to reproduce this for 2+ years. I'm pretty sure this is not related to grappelli but your server/django setup.

**Comment by ajostergaard** [comment]
It seems to me that the same symptom has multiple root causes depending on the environment.

In my case, 'runserver', the solution was to reverse STATICFILES_FINDERS as suggested above.

**Comment by ajostergaard** [comment]
ACTUALLY... In my case STATICFILES_FINDERS only partially solved the problem and, in fact, was a red herring.

What solved it for me was making sure grappelli loads before debug_toolbar:

INSTALLED_APPS = (
    ...
    'grappelli',
    'debug_toolbar',
    ...
)

**Comment by myii** [comment]
Have just encountered this problem for the second time so exerted some effort looking into it.  The following discussion is based upon my setup but it wouldn't surprise me in the slightest if others are in exactly the same boat.

>  sehmaschine commented on 11 Aug
> @sdillinger well, haven't been able to reproduce this for 2+ years. I'm pretty sure this is not related to grappelli but your server/django setup.

@sehmaschine: I'm with you 100% -- in my case at least, it had nothing to do with Grappelli itself.  As for reproducing (and fixing) the error -- perhaps I can help you with that...

---

##### TL;DR

A simple, working solution to this problem.  First up, two important references:

http://django-grappelli.readthedocs.org/en/latest/quickstart.html#setup:

> Open `settings.py` and add `grappelli` to your `INSTALLED_APPS` (before `django.contrib.admin`):
> 
> ```
> INSTALLED_APPS = (
>     'grappelli',
>     'django.contrib.admin',
> )
> ```

https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#django-admin-option---clear:
> `--clear`
> 
>    Clear the existing files before trying to copy or link the original file.

A caveat:

* Using `--clear` is like a sledgehammer; it will clear all static files that aren't needed by the current project
* So if you are using the same `STATIC_ROOT` for more than one project, then better to leave this method -- there's a more nuanced, less destructive method outlined [further down](#fixing_the_error)
* Essentially, **don't use this for anything other than testing**, unless you're comfortable to rebuild `STATIC_ROOT`

Now the steps:

1. Ensure you have `'grappelli'` before `'django.contrib.admin'` in `INSTALLED_APPS`, as shown in the reference above
1. Run `python manage.py collectstatic --noinput --clear`
1. Test from the browser, with full reload of the page with the filter button on it -- error should be resolved

---

##### Basic overview

###### Intro

With the steps shown below, I've been able to reproduce and fix the error 100% consistently, using the following setup:

1. Django 1.8
1. `DEBUG = False`
1. Nginx + Gunicorn via. Unix socket

In any case, as long as the static files are being accessed at `STATIC_ROOT`, the following steps should work.

###### Reproducing the error

In order to reproduce the error, note that I am about to *contradict the official instructions as shown above*, because that is one of the ways that this problem could have occurred.

* Switch the ordering in `INSTALLED_APPS`, i.e.:

```
INSTALLED_APPS = (
    'django.contrib.admin',
    'grappelli',
)
```

* Delete (or at least rename) the `static/admin/js` directory
* Run `python manage.py collectstatic --noinput`, output should show `django/contrib/admin` in the path, such as:

```
$ python manage.py collectstatic --noinput
Copying '.../django/contrib/admin/static/admin/js/inlines.js'
Copying '.../django/contrib/admin/static/admin/js/collapse.js'
Copying '.../django/contrib/admin/static/admin/js/related-widget-wrapper.js'
Copying '.../django/contrib/admin/static/admin/js/calendar.js'
Copying '.../django/contrib/admin/static/admin/js/collapse.min.js'
Copying '.../django/contrib/admin/static/admin/js/urlify.js'
Copying '.../django/contrib/admin/static/admin/js/core.js'
Copying '.../django/contrib/admin/static/admin/js/jquery.js'
Copying '.../django/contrib/admin/static/admin/js/actions.min.js'
Copying '.../django/contrib/admin/static/admin/js/jquery.init.js'
Copying '.../django/contrib/admin/static/admin/js/inlines.min.js'
Copying '.../django/contrib/admin/static/admin/js/actions.js'
Copying '.../django/contrib/admin/static/admin/js/jquery.min.js'
Copying '.../django/contrib/admin/static/admin/js/timeparse.js'
Copying '.../django/contrib/admin/static/admin/js/SelectBox.js'
Copying '.../django/contrib/admin/static/admin/js/prepopulate.min.js'
Copying '.../django/contrib/admin/static/admin/js/SelectFilter2.js'
Copying '.../django/contrib/admin/static/admin/js/LICENSE-JQUERY.txt'
Copying '.../django/contrib/admin/static/admin/js/prepopulate.js'
Copying '.../django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js'
Copying '.../django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js'

21 static files copied to '.../static', 593 unmodified.
```

* Test from the browser, with full reload of the page with the filter button on it -- error should be manifest

<a name="fixing_the_error"></a>
###### Fixing the error

The resolution steps are almost the same:

* Correct the ordering in `INSTALLED_APPS`, i.e.:

```
INSTALLED_APPS = (
    'grappelli',
    'django.contrib.admin',
)
```

* Delete (or at least rename) the `static/admin/js` directory
* Run `python manage.py collectstatic --noinput`, output should show `grappelli` in the path, such as:

```
$ python manage.py collectstatic --noinput
Copying '.../grappelli/static/admin/js/SelectFilter2.js'
Copying '.../grappelli/static/admin/js/collapse.min.js'
Copying '.../grappelli/static/admin/js/actions.min.js'
Copying '.../grappelli/static/admin/js/inlines.js'
Copying '.../grappelli/static/admin/js/urlify.js'
Copying '.../grappelli/static/admin/js/related-widget-wrapper.js'
Copying '.../grappelli/static/admin/js/collapse.js'
Copying '.../grappelli/static/admin/js/jquery.js'
Copying '.../grappelli/static/admin/js/jquery.min.js'
Copying '.../grappelli/static/admin/js/calendar.js'
Copying '.../grappelli/static/admin/js/prepopulate.js'
Copying '.../grappelli/static/admin/js/prepopulate.min.js'
Copying '.../grappelli/static/admin/js/actions.js'
Copying '.../grappelli/static/admin/js/core.js'
Copying '.../grappelli/static/admin/js/LICENSE-JQUERY.txt'
Copying '.../grappelli/static/admin/js/timeparse.js'
Copying '.../grappelli/static/admin/js/SelectBox.js'
Copying '.../grappelli/static/admin/js/inlines.min.js'
Copying '.../grappelli/static/admin/js/jquery.init.js'
Copying '.../grappelli/static/admin/js/admin/RelatedObjectLookups.js'
Copying '.../grappelli/static/admin/js/admin/DateTimeShortcuts.js'

21 static files copied to '.../static', 593 unmodified.
```

* Test from the browser, with full reload of the page with the filter button on it -- error should be resolved

###### Why is the problem occurring?

* `collectstatic` is getting the `static/admin/js` files from `'django.contrib.admin'` instead of `'grappelli'`

###### How could this have happened?

1. As in the steps shown above, could have the `INSTALLED_APPS` ordered wrongly
1. Another cause could be that `collectstatic` was run for `'django.contrib.admin'` before `'grappelli'` was ever installed
1. ...

---

##### Further details

The following points are from my basic investigations and not any meaningful experience.  I'd be perfectly happy for a Django expert to come along and tell me I'm all wrong and there are better explanations and solutions.  I just want to develop more robust, stable apps.  Please take with a pinch of salt.

###### Everything hinges on how `collectstatic` works

https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#collectstatic:

> On subsequent `collectstatic` runs (if `STATIC_ROOT` isn’t empty), **files are copied only if they have a modified timestamp greater than the timestamp of the file in `STATIC_ROOT`**. Therefore if you remove an application from `INSTALLED_APPS`, it’s a good idea to use the `--clear` option in order to remove stale static files.

* Emphasis above is mine
* Possible scenario:
    * Wrong `static/admin/js` in place
    * User has corrected ordering in `INSTALLED_APPS`
    * However, modification dates on system are unfavourable, i.e.:
        *  `grappelli/static/admin/js` < `static/admin/js`
* The result will be something like:

```
$ python manage.py collectstatic --noinput

0 static files copied to '.../static', 614 unmodified.
```

* Change the modification date for one of the source files, to show that it is then collected:

```
$ touch .../grappelli/static/admin/js/jquery.init.js 
$ python manage.py collectstatic --noinput
Copying '.../grappelli/static/admin/js/jquery.init.js'

1 static file copied to '.../static', 613 unmodified.
```

* Probably far more appropriate to change modification dates in the `static` directory rather than the source files, e.g.:

```
$ touch -d 2000-01-01 .../static/admin/js/jquery.init.js 
$ python manage.py collectstatic --noinput
Copying '.../grappelli/static/admin/js/jquery.init.js'

1 static file copied to '.../static', 613 unmodified.
```

* So for the whole `static/admin/js` directory:

```
$ find .../static/admin/js -type f -exec touch -d 2000-01-01 {} \;
$ python manage.py collectstatic --noinput
Copying '.../grappelli/static/admin/js/SelectFilter2.js'
Copying '.../grappelli/static/admin/js/collapse.min.js'
Copying '.../grappelli/static/admin/js/actions.min.js'
Copying '.../grappelli/static/admin/js/inlines.js'
Copying '.../grappelli/static/admin/js/urlify.js'
Copying '.../grappelli/static/admin/js/related-widget-wrapper.js'
Copying '.../grappelli/static/admin/js/collapse.js'
Copying '.../grappelli/static/admin/js/jquery.js'
Copying '.../grappelli/static/admin/js/jquery.min.js'
Copying '.../grappelli/static/admin/js/calendar.js'
Copying '.../grappelli/static/admin/js/prepopulate.js'
Copying '.../grappelli/static/admin/js/prepopulate.min.js'
Copying '.../grappelli/static/admin/js/actions.js'
Copying '.../grappelli/static/admin/js/core.js'
Copying '.../grappelli/static/admin/js/LICENSE-JQUERY.txt'
Copying '.../grappelli/static/admin/js/timeparse.js'
Copying '.../grappelli/static/admin/js/SelectBox.js'
Copying '.../grappelli/static/admin/js/inlines.min.js'
Copying '.../grappelli/static/admin/js/jquery.init.js'
Copying '.../grappelli/static/admin/js/admin/RelatedObjectLookups.js'
Copying '.../grappelli/static/admin/js/admin/DateTimeShortcuts.js'

21 static files copied to '.../static', 593 unmodified.
```

* In regards to the `--clear` option, as mentioned at the beginning; some observations:
    * It definitely works...
    * ... but it is a very aggressive solution, which can end up removing static files belonging to other projects that share the same `STATIC_ROOT`
    * Thus, not really appropriate as a general recommendation

###### Repercussions

This issue remaining active for over 3 years is a testament to this `collectstatic` functionality.  What makes it even more delicious is that the "bug" can resolve itself given the right conditions.  An example:

* Amongst trying different solutions, user upgrades Grappelli, introducing new `static/admin/js` source files, with more recent modification dates than those currently in `STATIC_ROOT`
    * Yet, filter button stubbornly refuses to work
* At some future point in time, user runs `collectstatic`
    * Suddenly, the filter button is working again

In an ideal world:

* An in-between setting for `collectstatic`, something like a `--replace` or `--overwrite` or `--ignore-timestamps`
* Basically:
    * Find all the static files as it currently does
    * Copy them over regardless of existing static files
    * Don't touch any other static files in `STATIC_ROOT`

Now if this ideal world already exists, please let me know.  Perhaps even this issue could finally be closed!

###### "The File of Doom"

Finally, this whole discussion wouldn't be complete without identifying the offender, fingering the exact line of code that has caused so much pain!

I've already sneakily mentioned it above... and it is...:

* **`static/admin/js/jquery.init.js`**!!!  But y'all had already figured that out, right?!

If Grappelli wins the `collectstatic` battle, then you'll find it contains:

```
// dropped
// not used in grappelli
// kept this file to prevent 404
```

Yet if Django-Admin comes out trumps, you'll find:

```
/* Puts the included jQuery into our own namespace using noConflict and passing
 * it 'true'. This ensures that the included jQuery doesn't pollute the global
 * namespace (i.e. this preserves pre-existing values for both window.$ and
 * window.jQuery).
 */
var django = django || {};
django.jQuery = jQuery.noConflict(true);
```

* So, just a 2-line `js` file after all of that
* And it's actually the last line that is the troublemaker -- comment it out and the filter button works again

NB: I am in **no way** advocating this method of getting the filter button working again -- just sharing this finding for the curious out there.

**Comment by sehmaschine** [comment]
@myii Thanks a lot for the research and the very detailed explanation. What you're saying makes sense indeed (still funny that I've never run across this issue with any projects we did).

**Comment by Tatsh** [comment]
@myii Thanks. I added the steps` to my deployment in Ansible like so, and then run collectstatic again (unfortunately):

```yaml
- name: 'manage.py collectstatic'
  django_manage: app_path={{common_htdocs_dir}}/fos
                 command=collectstatic
                 settings=fos.settings
                 virtualenv={{common_htdocs_dir}}/fos-venv
  when: code.changed or code_deps.changed or settings.changed
  tags:
    - fos
    - collectstatic

# Next two steps, credits to @myii
# https://github.com/sehmaschine/django-grappelli/issues/214#issuecomment-158576652
- name: "Move Django's admin assets to avoid conflict with Grapelli's"
  command: mv {{common_htdocs_dir}}/fos/static/admin/js {{common_htdocs_dir}}/fos/static/admin/js-already-moved creates={{common_htdocs_dir}}/fos/static/admin/js-already-moved
  tags:
    - fos
    - collectstatic
    - grapelli

# Using command directly so ----no-post-process can be passed, for extra speed
- name: "Re-run collectstatic to replace admin assets with Grapelli's"
  command: '{{common_htdocs_dir}}/fos-venv/bin/python ./manage.py collectstatic --no-color --noinput --no-post-process chdir={{common_htdocs_dir}}/fos'
  tags:
    - fos
    - collectstatic
```

Luckily Grapelli does not need post-processing, and only JS is affected.

**Comment by fchevitarese** [comment]
I got the same error, added this and it worked
STATICFILES_FINDERS = (
'django.contrib.staticfiles.finders.AppDirectoriesFinder',
'django.contrib.staticfiles.finders.FileSystemFinder',
}

;)

**Comment by marcoooo** [comment]
Hi all, Django 1.9.8 / Grappelli 2.8., this problem still occurs, even in production with static files setup. 

Is there any 'simple' solution for this ?

**Comment by ecederstrand** [comment]
marcoooo,

Is `grappelli` first in `INSTALLED_APPS`?

```
INSTALLED_APPS = (
    'grappelli',
    'django.contrib.admin',
)
```

Did you run `collectstatic`? What is the output? Which filesystem path does it collect to? How are your static files served? Which filesystem path are they served from? Have you emptied your browser cache and any server caches for your static files?

**Comment by myii** [comment]
@marcoooo Based upon [my long answer above](https://github.com/sehmaschine/django-grappelli/issues/214#issuecomment-158576652) and the [follow-up by @Tatsh](https://github.com/sehmaschine/django-grappelli/issues/214#issuecomment-159800219), I've got a 2-step summary for you:

1. Clear/move the `static/admin/js` directory
1. Run `collectstatic` again

As long as you've got the correct ordering as pointed out by @ecederstrand, you'll be good to go.

**Comment by marcoooo** [comment]
Hi All, thank for your quiclk response !
- Ordering in apps is ok (since in the reverse order, I d'ont get the grappelli style at all)
- I run collectstatic, admin files comme again (so I deleted the folder)
- path is (relative to project ~/staticfiles/) I think my conf is correct because all other statics are served well (no urlpatterns update in my urls.py configuration, as recommanded in Django doc)

Still got this issue.

Django packages installed:

> django-admin-sortable (2.0.18)
django-authtools (1.5.0)
django-bootstrap-themes (3.3.6)
django-braces (1.9.0)
django-ckeditor (5.1.0)
django-cors-headers (1.1.0)
django-countries (3.4.1)
django-crispy-forms (1.6.0)
django-crontab (0.7.1)
django-debug-toolbar (1.5)
django-eav (0.9.2, /home/marc/.virtualenvs/waves1/src/django-eav)
django-environ (0.4.0)
django-grappelli (2.8.1)
django-ipware (1.1.5)
django-jquery (1.12.2)
django-jquery-ui (1.11.4.1)
django-log-file-viewer (0.9)
django-mail-templated (2.6.2)
django-mptt (0.8.4)
django-multiupload (0.5.1)
django-nested-admin (3.0.8)
django-polymorphic (1.0.1)
django-registration (2.1.2)
django-smart-selects (1.2.2)
django-tabbed-admin (1.0.0)
djangorestframework (3.4.0)
djangorestframework-jwt (1.8.0)
djangorestframework-xml (1.3.0)
drfdocs (0.0.11)

Running on apache with static serving setup:

 &lt;Directory [ABSOLUTE_PATH_TO_WAVES_INSTALL_DIR]/src/waves_services&gt;
        Options FollowSymLinks Indexes
        &lt;Files wsgi.py&gt;
	        Require all granted
        &lt;/Files&gt;
    &lt;/Directory&gt;
    &lt;Location /static&gt;
      SetHandler None
    &lt;/Location&gt;
    &lt;Location /media&gt;
      SetHandler None
    &lt;/Location&gt;
    Alias /media [ABSOLUTE_PATH_TO_WAVES_INSTALL_DIR]/media
    Alias /static [ABSOLUTE_PATH_TO_WAVES_INSTALL_DIR]/staticfiles

**Comment by myii** [comment]
> I run collectstatic, admin files comme again (so I deleted the folder)

@marcoooo Only delete the folder **before** `collectstatic`.  When you run `collectstatic` again, it produces the correct files *that you need to keep*.

**Comment by tramasoli** [comment]
Hello everyone!
We have here the same issue but with a different scenario:
- 2 virtualenvs packaged with RPM
- one of them have Grappelli
our static files are shared across the virtualenv and collectstatic is run first for the one that do not have Grappelli.
The timestamp of files are the date of the RPM's creation, so Grappellis and Djando static admin files are almost ate the same datetime.

If someone else got this scenario beware. Our solution (for now) is to mannualy copy Grappelli's static files after collectstatic.

**Comment by sehmaschine** [comment]
Closing this since the issue is extremely outdated. If there is still something unresolved related to this topic, please open a new ticket.

**Comment by MahrezBH** [comment]
I faced this issue, which I believe may be due to compatibility problems with `django-material` or another frontend package. I tried all the above suggestions to fix it, but without success.

The only solution I found is to re-check the `actions.js` file (located either in your `static/admin/js` directory or the `django/contrib/admin/static/admin/js` directory)

The `actions.js` file:
```javascript
```
(function($) {
	var lastChecked;

	$.fn.actions = function(opts) {
		var options = $.extend({}, $.fn.actions.defaults, opts);
		var actionCheckboxes = $(this);
		var list_editable_changed = false;
		var checker = function(checked) {
			if (checked) {
				showQuestion();
			} else {
				reset();
			}
			$(actionCheckboxes).prop("checked", checked)
				.parent().parent().toggleClass(options.selectedClass, checked);
		},
		updateCounter = function() {
			var sel = $(actionCheckboxes).filter(":checked").length;
			// _actions_icnt is defined in the generated HTML
			// and contains the total amount of objects in the queryset
			$(options.counterContainer).html(interpolate(
			ngettext('%(sel)s of %(cnt)s selected', '%(sel)s of %(cnt)s selected', sel), {
				sel: sel,
				cnt: _actions_icnt
			}, true));
			$(options.allToggle).prop("checked", function() {
				var value;
				if (sel == actionCheckboxes.length) {
					value = true;
					showQuestion();
				} else {
					value = false;
					clearAcross();
				}
				return value;
			});
		},
		showQuestion = function() {
			$(options.acrossClears).hide();
			$(options.acrossQuestions).show();
			$(options.allContainer).hide();
		},
		showClear = function() {
			$(options.acrossClears).show();
			$(options.acrossQuestions).hide();
			$(options.actionContainer).toggleClass(options.selectedClass);
			$(options.allContainer).show();
			$(options.counterContainer).hide();
		},
		reset = function() {
			$(options.acrossClears).hide();
			$(options.acrossQuestions).hide();
			$(options.allContainer).hide();
			$(options.counterContainer).show();
		},
		clearAcross = function() {
			reset();
			$(options.acrossInput).val(0);
			$(options.actionContainer).removeClass(options.selectedClass);
		};
		// Show counter by default
		$(options.counterContainer).show();
		// Check state of checkboxes and reinit state if needed
		$(this).filter(":checked").each(function(i) {
			$(this).parent().parent().toggleClass(options.selectedClass);
			updateCounter();
			if ($(options.acrossInput).val() == 1) {
				showClear();
			}
		});
		$(options.allToggle).show().click(function() {
			checker($(this).prop("checked"));
			updateCounter();
		});
		$("a", options.acrossQuestions).click(function(event) {
			event.preventDefault();
			$(options.acrossInput).val(1);
			showClear();
		});
		$("a", options.acrossClears).click(function(event) {
			event.preventDefault();
			$(options.allToggle).prop("checked", false);
			clearAcross();
			checker(0);
			updateCounter();
		});
		lastChecked = null;
		$(actionCheckboxes).click(function(event) {
			if (!event) { event = window.event; }
			var target = event.target ? event.target : event.srcElement;
			if (lastChecked && $.data(lastChecked) != $.data(target) && event.shiftKey === true) {
				var inrange = false;
				$(lastChecked).prop("checked", target.checked)
					.parent().parent().toggleClass(options.selectedClass, target.checked);
				$(actionCheckboxes).each(function() {
					if ($.data(this) == $.data(lastChecked) || $.data(this) == $.data(target)) {
						inrange = (inrange) ? false : true;
					}
					if (inrange) {
						$(this).prop("checked", target.checked)
							.parent().parent().toggleClass(options.selectedClass, target.checked);
					}
				});
			}
			$(target).parent().parent().toggleClass(options.selectedClass, target.checked);
			lastChecked = target;
			updateCounter();
		});
		$('form#changelist-form table#result_list tr').find('td:gt(0) :input').change(function() {
			list_editable_changed = true;
		});
		$('form#changelist-form button[name="index"]').click(function(event) {
			if (list_editable_changed) {
				return confirm(gettext("You have unsaved changes on individual editable fields. If you run an action, your unsaved changes will be lost."));
			}
		});
		$('form#changelist-form input[name="_save"]').click(function(event) {
			var action_changed = false;
			$('select option:selected', options.actionContainer).each(function() {
				if ($(this).val()) {
					action_changed = true;
				}
			});
			if (action_changed) {
				if (list_editable_changed) {
					return confirm(gettext("You have selected an action, but you haven't saved your changes to individual fields yet. Please click OK to save. You'll need to re-run the action."));
				} else {
					return confirm(gettext("You have selected an action, and you haven't made any changes on individual fields. You're probably looking for the Go button rather than the Save button."));
				}
			}
		});
	};
	/* Setup plugin defaults */
	$.fn.actions.defaults = {
		actionContainer: "div.actions",
		counterContainer: "span.action-counter",
		allContainer: "div.actions span.all",
		acrossInput: "div.actions input.select-across",
		acrossQuestions: "div.actions span.question",
		acrossClears: "div.actions span.clear",
		allToggle: "#action-toggle",
		selectedClass: "selected"
	};
})(django.jQuery);
```
This `actions.js` file contains the default actions handling script used by Django's admin interface to manage multiple selection checkboxes, ensuring proper handling of selected actions and form submissions. If you're still facing the issue, consider investigating other JS files in your setup that could be conflicting, or try updating your frontend packages to the latest compatible versions with Django.
